WordPress/Classes/ViewRelated/Domains/Register/RegisterDomainDetails/ViewController/RegisterDomainDetailsViewController.swift

```diff
@@ -0,0 +1,362 @@
+        let gestureRecognizer = UITapGestureRecognizer()
+        gestureRecognizer.cancelsTouchesInView = false
+
+        gestureRecognizer.on { [weak self] (gesture) in
+            self?.view.endEditing(true)
+        }
+
+        view.addGestureRecognizer(gestureRecognizer)
+    }
+
+    private func handle(change: RegisterDomainDetailsViewModel.Change) {
+        switch change {
+        case let .formValidated(context, isValid):
+            switch context {
+            case .clientSide:
+                footerView.submitButton.isEnabled = isValid
+            default:
+                break
+            }
+            break
+        case .addNewAddressLineEnabled(let indexPath):
+            tableView.insertRows(at: [indexPath], with: .none)
+        case .addNewAddressLineReplaced(let indexPath):
+            tableView.reloadRows(at: [indexPath], with: .none)
+        case .checkMarkRowsUpdated:
+            tableView.reloadData()
+        case .registerSucceeded(let items):
+            //TODO: temporarily show as an alert
+            showAlert(title: "Success", message: items.description)
+        case .unexpectedError(let message):
+            showAlert(message: message)
+        case .loading(let isLoading):
+            if isLoading {
+                SVProgressHUD.show()
+            } else {
+                SVProgressHUD.dismiss()
+            }
+        case .prefillSuccess:
+            tableView.reloadData()
+        case .prefillError(let message):
+            showAlert(message: message)
+        case .multipleChoiceRowValueChanged(let indexPath):
+            tableView.reloadRows(at: [indexPath], with: .none)
+        case .remoteValidationFinished:
+            tableView.reloadData()
+        default:
+            break
+        }
+    }
+
+}
+
+// MARK: - Actions
+
+extension RegisterDomainDetailsViewController {
+
+    @objc private func registerDomainButtonTapped(sender: UIButton) {
+        viewModel.register()
+    }
+
+    @objc func handleTermsAndConditionsTap(_ sender: UITapGestureRecognizer) {
+        UIApplication.shared.open(URL(string: WPAutomatticTermsOfServiceURL)!, options: [:], completionHandler: nil)
+    }
+
+}
+
+// MARK: - InlineEditableNameValueCellDelegate
+
+extension RegisterDomainDetailsViewController: InlineEditableNameValueCellDelegate {
+
+    func inlineEditableNameValueCell(_ cell: InlineEditableNameValueCell,
+                                     valueTextFieldDidChange valueTextField: UITextField) {
+        guard let indexPath = tableView.indexPath(for: cell),
+            let sectionType = SectionIndex(rawValue: indexPath.section) else {
+                return
+        }
+        viewModel.updateValue(valueTextField.text, at: indexPath)
+
+        switch sectionType {
+        case .address:
+            let addressField = viewModel.addressSectionIndexHelper.addressField(for: indexPath.row)
+            switch addressField {
```

Comment:
I think you inherited this, but the switches on `sectionType` and `addressField` seem unnecessary, and makes this overly complex. Can `if`s or `guard`s be used to streamline this?